### PR TITLE
Support publishing legacy-events for Subscritpion v1alpha2 CRD in EPP

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added support publishing legacy-events for Subscritpion v1alpha2 CRD in EPP.
- Replaced hardcoded old Event type prefix in EPP.

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/15618

### CODE REVIEW

**Review by nachtmaar — COMMENTED** [review]
Mostly nitpicks :-D

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:334]: We don't check it explicitly but in some test cases we are returning an error from the sender and expect the request fails. 

For example, in test case:`Send valid legacy event but cannot send to backend due to target not found (e.g. stream missing)`

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:346]: This mock is used by metrics collector. The metrics collector is tested at the end of each test case (`metricstest.EnsureMetricMatchesTextExpositionFormat(t, h.collector, tt.wantTEF)`), so this would be covered. 

Also, latency mock is defined globally for all test cases, so doing `latency.AssertExpectations(t)` would be green even if its called by any single test case. To test it per test case, we have to define a new latency mock per test case and check `latency.AssertExpectations(t)`.

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:347]: According to [docs](https://pkg.go.dev/github.com/stretchr/testify/mock#Mock.Test), it just sets the testing struct in mock. It was not useful so I have removed it. 

It is not an alternative to `latency.AssertExpectations(t)`.

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:368]: `eventTypeCleaner` was  not required so I removed it. `collector` is required by metrics tests. Anyway I have removed the `fields` and `args` and using simple `givenXxxx` variables to make it simpler.

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:493]: lets do this later in a separate PR once this is merged.

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:495]: yes, but this is used in other places as well and I dont want to add alot of changes in this PR. I can do it in a separate PR once this is merged.

File: components/event-publisher-proxy/pkg/handler/handler.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler.go:134]: updated

File: components/event-publisher-proxy/pkg/handler/handler.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler.go]: updated

File: components/event-publisher-proxy/pkg/handler/handler.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler.go]: updated

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go]: updated

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go]: updated

File: components/event-publisher-proxy/pkg/legacy/api/types.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/api/types.go]: updated

File: components/event-publisher-proxy/pkg/legacy/helpers.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/helpers.go]: updated

File: components/event-publisher-proxy/pkg/legacy/legacy.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy.go]: updated

File: components/event-publisher-proxy/pkg/legacy/legacy.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy.go]: I moved it to `ExtractPublishRequestData` where rest of the validation was done. It just validating if the application name is clean or not. It does not clean it. The behaviour was the same in old legacy endpoint so I just kept it as it is.

File: components/event-publisher-proxy/pkg/legacy/legacy.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy.go]: updated the name

File: components/event-publisher-proxy/pkg/legacy/legacy_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy_test.go]: updated

File: components/event-publisher-proxy/pkg/legacy/legacy_test.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy_test.go]: updated

File: components/event-publisher-proxy/pkg/legacy/legacy.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/legacy/legacy.go:30]: Well, its only for legacy endpoint and a single method. IMO, having a separate struct would be overkill for now, then we would have to initialise it in commander and pass it to handler etc.

File: components/event-publisher-proxy/pkg/handler/handler.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/handler/handler.go]: updated

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go]: Ok, just asking because I saw it recently and did not understand the benefits :-D

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:456]: yeah sure. Just wanted to exchange some information :)

Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go:456]: Was not intended as a call to action : )

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go]: Doesn't that mean that exactly the happy path is not covered but only the error cases ? An easy interaction test with a mock could cover this test. I leave it up to you if you wanna implement it :)

File: components/event-publisher-proxy/pkg/legacy/legacy.go
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/legacy/legacy.go:30]: I see :)

File: components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go]: Discussed in person.

File: components/event-publisher-proxy/pkg/env/types.go
Comment (nachtmaar) [inline components/event-publisher-proxy/pkg/env/types.go:7]: nit: You can use the ActiveBackend typedef here directly :)

```suggestion
        JetStreamBackend ActiveBackend = "JetStream"
	EventMeshBackend ActiveBackend = "EventMesh"
```

Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/env/types.go:7]: Shouldn’t this be inside `pkg/backend/jetstreamv2` instead ? It would then be `jetstreamv2.ActiveBackend` instead of `env.ActiveBackend`

**Review by nachtmaar — APPROVED** [review]
LGTM. Thanks for your contribution and nicely documented code :)

File: components/event-publisher-proxy/pkg/env/types.go
Comment (mfaizanse) [reply components/event-publisher-proxy/pkg/env/types.go:7]: ideally, but this will be cleaned up after the deprecation of v1alpha1 code base.

**Review by nachtmaar — APPROVED** [review]

**Pull request merged**
